### DISCUSSION TITLE
Two dependency updates are stuck because they require each others update

### DISCUSSION THREAD

**Discussion by dahaupt** [renovatebot/renovate#10963] [Request Help] [answered]
In our PHP/Composer project there is an update for two packages. Currently, both packages cannot be updated, because they need exactly the update of the other package as requirement.

**Update 1: Package `rector/rector` to v0.4.0**
Artifact update problem (from composer command):
`rector/rector 0.11.40 requires phpstan/phpstan 0.12.93 -> found phpstan/phpstan[0.12.93] but it conflicts with your root composer.json require (0.12.91).`

**Update 2: Package `phpstan/phpstan` to v0.12.93**
Artifact update problem (from composer command):
`rector/rector 0.11.38 requires phpstan/phpstan 0.12.91 -> found phpstan/phpstan[0.12.91] but it conflicts with your root composer.json require (0.12.93).`

In the past, I solved those conflicts manually. Is it possible to solve the problem with the existing configuration options or is the case so rare that it is not worth covering?

**Reproduction**: https://github.com/dahaupt/renovate-composer-tests

**Comment by dahaupt** [comment, marked as answer]
Could this be solved via a packageRule entry?

**Reply by rarkins** [reply]
Yes, you can apply grouping if dependencies are co-dependent like this. A `packageRules` entry with `matchPackageNames` and `groupName`.

**Reply by dahaupt** [reply]
@rarkins Thanks for your fast reply. I have created the necessary [packageRule entry](https://github.com/dahaupt/renovate-composer-tests/blob/main/renovate.json#L7-L11), which works fine.

I had also thought about whether Renovate should be able to detect these situations and then create an automatic "temporary" grouping to make the update work. What do you think about this idea? But probably this conflict situation is too rare.
